Add coverage to file_regex status check method

braeeast2001 opened the pull request:
Add test_file_regex to test_builders.py using regex_on_filename.yml.

braeeast2001 committed:
Update test_builders.py to add file_regex status check

szuananwar commented on tests/builders/test_builders.py:
```diff
@@ -138,3 +138,15 @@ def test_file_count():
+    """This test buildspec using status check with 'file_regex'"""
+    cmd = BuildTest(
+        buildspecs=[
+            os.path.join(here, "file_regex.yml")
```
line 146: These is no file_regex.yml file in tutorials/test_status to link to, please make sure the name of yml file is correct.

szuananwar reviewed (commented):
line 146: These is no file_regex.yml file in tutorials/test_status to link to, please make sure the name of yml file is correct.

braeeast2001 committed:
Update test_builders.py

Corrected file name

szuananwar commented on tests/builders/test_builders.py:
```diff
@@ -143,7 +143,7 @@ def test_file_regex():
     cmd = BuildTest(
         buildspecs=[
-            os.path.join(here, "file_regex.yml")
+            os.path.join(here, "status_regex.yml")
```
Use this test https://github.com/buildtesters/buildtest/blob/devel/tutorials/test_status/regex_on_filename.yml for your regression test.

braeeast2001 committed:
Update test_builders.py

Corrected file name

szuananwar commented on tests/builders/test_builders.py:
```diff
@@ -138,3 +138,15 @@ def test_file_count():
     cmd.build()
+
+def test_file_regex():
+    """This test buildspec using status check with 'file_regex'"""
```
Please change the comment here to match with the correct yml file name that will be linked for test.

szuananwar reviewed (commented):
Please change the comment in 143 to match with the correct yml file name that will be linked for test.

braeeast2001 committed:
Update test_builders.py

Fixed comment in line 143

szuananwar reviewed (commented):
The comment in the methods matches the linked yml file.

shahzebsiddiqui commented on tests/builders/test_builders.py:
```diff
@@ -138,3 +138,15 @@ def test_file_count():
     cmd.build()
+
+def test_file_regex():
+    """This test buildspec using status check with 'regex_on_filename'"""
```
So this docstring is not correct. The filename is `regex_on_filename` but that not what the test is doing. The test is performing status check on keyword `file_regex`. See https://buildtest.readthedocs.io/en/devel/buildspecs/buildspec_overview.html#specify-regular-expression-on-arbitrary-filenames

I would rephrase this docstring to 

```python
def test_file_regex():
    """This test will perform status check with regular expression on file via 'file_regex'"""
```

szuananwar replied:
@shahzebsiddiqui, the filename is regex_on_filename and the test is performing on regex_on_filename keyword. She got it and I commented "The comment in the methods matches the linked yml file" 14 hours ago.

shahzebsiddiqui replied:
Yes what you said is correct but the docstring makes no sense it is too technical and tied to the name of file and test name which can change. The actual keyword that is being tested is `file_regex` that is used in buildspec to specify a filename to perform regular expression. I think even removing the key word `file_regex` would be sufficient too

szuananwar replied:
Okay, thanks,
@braeeast2001 please see the above comment by @shahzebsiddiqui and update your docstring.

shahzebsiddiqui committed:
update docstring for the test to be more meaningful to the what is being tested

shahzebsiddiqui committed:
Merge branch 'devel' into Brae'zah_Patch

braeeast2001 committed:
Create symlink and update line 143

braeeast2001 committed:
Merge branch 'Brae'zah_Patch' of https://github.com/braeeast2001/buildtest into Brae'zah_Patch

braeeast2001 committed:
Delete regex_on_filename.yml

braeeast2001 committed:
Update test_builders.py

braeeast2001 committed:
Merge branch 'Brae'zah_Patch' of https://github.com/braeeast2001/buildtest into Brae'zah_Patch

braeeast2001 committed:
Update test_builders.py

szuananwar commented on tests/builders/test_builders.py:
```diff
@@ -140,11 +140,10 @@ def test_file_count():
 
-def test_regex_check():
-    """This test buildspec using status check with  'status_regex'"""
+def test_file_regex():
```
You deleted Test_regex_check method, you suppose to add your method and not delete the exist one.

braeeast2001 committed:
Re-add Dr. Anwar method

braeeast2001 committed:
Correct Dr. A test name

braeeast2001 committed:
Create NEW file and Update test_builders.py

braeeast2001 committed:
Add New lien at the END of the file

braeeast2001 committed:
Merge branch 'devel' into Brae'zah_Patch

shahzebsiddiqui commented on tutorials/test_status/regex_on_invailds.yml:
```diff
@@ -0,0 +1,11 @@
+    status:
+      file_regex:
+      - file: /fox.txt
+      - exp: '^(Hello)$'
```
@braeeast2001 good job this looks good now can you add one more test in `regex_on_invalids.yml` that will perform regular expression on valid file but regex is invalid.

Also there is a spelling error in name of file `regex_on_invailds.yml` -> `regex_on_invalids.yml`

shahzebsiddiqui commented on tests/builders/test_builders.py:
```diff
@@ -151,9 +151,20 @@ def test_multicompilers_with_script_schema():
 
 def test_regex_check():
-    """This test buildspec using status check with  'status_regex'"""
+     """This test buildspec using status check with  'status_regex'"""
```
change `status_regex` -> `regex` in doc string

braeeast2001 committed:
Update regex_on_invalids.yml

Create new test for valid file_name with invalid regex and correct spelling mistake

braeeast2001 committed:
Merge branch 'Brae'zah_Patch' of https://github.com/braeeast2001/buildtest into Brae'zah_Patch

shahzebsiddiqui commented on tutorials/test_status/regex_on_invalids.yml:
```diff
@@ -0,0 +1,22 @@
+      - file: /fox.txt
+      - exp: '^(Hello)$'
+
+regex_where_validfname_invaildregex:
```
spelling error here can you change the name `file_regex_with_invalid_expression`

shahzebsiddiqui commented on tutorials/test_status/regex_on_invalids.yml:
```diff
@@ -0,0 +1,22 @@
+build specs:
```
this is invalid buildspec its `build specs` -> `buildspecs`

shahzebsiddiqui commented on tutorials/test_status/regex_on_invalids.yml:
```diff
@@ -0,0 +1,22 @@
+    echo "Jump" > jump.txt
+  status:
+    file_regex:
+      - file: foo/bar.txt
```
change this to `jump.txt` we want the file to be valid here

shahzebsiddiqui commented on tutorials/test_status/regex_on_invalids.yml:
```diff
@@ -0,0 +1,22 @@
+  executor: generic.local.bash
+  description: Test regex on a valid file where regex is invaild
+  run: |
+    echo "Jump" > jump.txt
```
please put this on a single line `run: echo "Jump" > jump.txt`

If its multi line then do what you did

braeeast2001 committed:
Correct spelling mistakes

szuananwar reviewed (commented):
Thanks @braeeast2001 for making all the necessary changes. 
The style check is failed, please run buildtest sytle -a
and see this for more info: https://buildtest.readthedocs.io/en/devel/contributing/code_contribution_guide.html

shahzebsiddiqui committed:
fix issue with test it must be in the tests/builders directory
and buildspec was invalid

shahzebsiddiqui committed:
fix style check issue

szuananwar commented on tests/builders/test_builders.py:
```diff
@@ -151,21 +151,22 @@ def test_multicompilers_with_script_schema():
-                    os.path.join(here, "regex_on_invalids.yml")],
+        buildspecs=[
+            os.path.join(here, "regex_on_filename.yml"),
+            os.path.join(here, "regex_on_invalids.yml"),
```
Please delete the comma , before ]

shahzebsiddiqui committed:
fix issue with yamllint
